File: languages/python/bloock/entity/identity_v2/bjj_identity_key.py
import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.identity_key_args import IdentityKeyArgs
from bloock.entity.identity_v2.identity_key import IdentityKey


class BjjIdentityKey(IdentityKey):

    def __init__(self, args: IdentityKeyArgs) -> None:
        self.args = args

    def to_proto(self) -> proto.IdentityKey:
        local_key = None
        if self.args.local_key is not None:
            local_key = self.args.local_key.to_proto()

        managed_key = None
        if self.args.managed_key is not None:
            managed_key = self.args.managed_key.to_proto()

        return proto.IdentityKey(
            local_key=local_key,
            managed_key=managed_key,
        )

File: languages/js/src/entity/identity_v2/bjj_identity_key.ts
import { IdentityKey as IdentityKeyProto } from "../../bridge/proto/identity_entities_v2";
import { LocalKey, ManagedKey } from "../key";
import { IdentityKey } from "./identity_key";
import { IdentityKeyArgs } from "./identity_key_args";

/**
 * Represents an identity BJJ key used.
 */
export class BjjIdentityKey implements IdentityKey {
  args: IdentityKeyArgs;

  /**
   * Creates a new BjjIdentityKey instance with the provided issuer key arguments.
   * @param key 
   */
  constructor(key: LocalKey | ManagedKey) {
    this.args = new IdentityKeyArgs(key);
  }

  public toProto(): IdentityKeyProto {
    return IdentityKeyProto.fromPartial({
      localKey: this.args.localKey?.toProto(),
      managedKey: this.args.managedKey?.toProto()
    });
  }
}
